File: backend/app/schemas/ai.py
"""Pydantic schemas for the AI query endpoints — Phase 6.3."""

from datetime import datetime
from typing import Any
from pydantic import BaseModel, Field


class AiQueryRequest(BaseModel):
    question: str = Field(..., min_length=3, max_length=1000,
                          description="Natural-language question about marketing data")


class AiQueryResponse(BaseModel):
    id: str
    question: str
    pipeline: list[dict[str, Any]]
    result: list[dict[str, Any]]
    explanation: str
    created_at: str  # ISO-8601


class AiHistoryItem(BaseModel):
    id: str
    question: str
    explanation: str
    result_count: int
    created_at: str  # ISO-8601


class AiHistoryResponse(BaseModel):
    items: list[AiHistoryItem]
    total: int
    limit: int
    offset: int

File: frontend/types/ai.ts
// ── AI query types — Phase 6.4 ────────────────────────────────────────────────

export interface AiQueryResponse {
  id: string;
  question: string;
  pipeline: Record<string, unknown>[];
  result: Record<string, unknown>[];
  explanation: string;
  created_at: string;
}

export interface AiHistoryItem {
  id: string;
  question: string;
  explanation: string;
  result_count: number;
  created_at: string;
}

export interface AiHistoryData {
  items: AiHistoryItem[];
  total: number;
  limit: number;
  offset: number;
}
